HPXMLtoOpenStudio/resources/hvac.rb

```diff
@@ -3555,6 +3570,110 @@ def self.apply_installation_quality(model, heating_system, cooling_system, unita
     program_calling_manager.addProgram(fault_program)
   end
 
+  def self.calculate_heat_pump_defrost_load_power_watts(heat_pump, unit_multiplier, design_airflow, max_heating_airflow, fan_watts_per_cfm)
+    return [nil, nil] unless heat_pump.advanced_defrost_approach
+
+    defrost_flow_fraction = max_heating_airflow / design_airflow
+    defrost_power_fraction = defrost_flow_fraction**3
+    power_design = fan_watts_per_cfm * design_airflow
+    p_dot_blower = power_design * defrost_power_fraction
+    rated_clg_capacity = UnitConversions.convert(heat_pump.cooling_capacity, 'Btu/hr', 'W') / unit_multiplier
+    p_dot_odu_fan = 44.348 * UnitConversions.convert(heat_pump.cooling_capacity, 'Btu/hr', 'ton') + 62.452
+    rated_clg_cop = heat_pump.additional_properties.cool_rated_cops[-1] # Fixme: assume highest stage cop?
+    q_dot_defrost = rated_clg_capacity * 0.45 # defrost cooling rate, 0.45 is from Jon's lab and field data analysis, defrost is too short to reach steady state so using cutler curve is not correct
+    cop_defrost = rated_clg_cop / 1.0 # defrost cooling cop, 1.0 is from Jon's lab and field data analysis, defrost is too short to reach steady state so using cutler curve is not correct
+    p_dot_defrost = (q_dot_defrost / cop_defrost - p_dot_odu_fan + p_dot_blower) * unit_multiplier # p_dot_defrost is used in coil object, which needs to be scaled up for unit multiplier
+    return q_dot_defrost, p_dot_defrost
+  end
+
+  def self.apply_advanced_defrost(model, htg_coil, air_loop_unitary, conditioned_space, htg_supp_coil, heat_pump, q_dot_defrost)
+    if htg_supp_coil.nil?
+      backup_system = heat_pump.backup_system
+      if backup_system.nil?
+        supp_sys_capacity = 0.0
+        supp_sys_power_level = 0.0
+        supp_sys_fuel = HPXML::FuelTypeElectricity
+      else
+        supp_sys_fuel = backup_system.heating_system_fuel
+        supp_sys_capacity = UnitConversions.convert(backup_system.heating_capacity, 'Btu/hr', 'W')
+        supp_sys_efficiency = backup_system.heating_efficiency_percent
+        supp_sys_efficiency = backup_system.heating_efficiency_afue if supp_sys_efficiency.nil?
+        supp_sys_power_level = [supp_sys_capacity, q_dot_defrost].min / supp_sys_efficiency # Assume perfect tempering
+      end
+    else
+      supp_sys_fuel = heat_pump.backup_heating_fuel
+      supp_sys_capacity = UnitConversions.convert(heat_pump.backup_heating_capacity, 'Btu/hr', 'W')
+      supp_sys_efficiency = heat_pump.backup_heating_efficiency_percent
+      supp_sys_efficiency = heat_pump.backup_heating_efficiency_afue if supp_sys_efficiency.nil?
+      supp_sys_power_level = [supp_sys_capacity, q_dot_defrost].min / supp_sys_efficiency # Assume perfect tempering
```

Comment:
Probably set supp_sys_power_level to 0.0 for ductless system, practically there will be no integrated supplemental system for ductless. @jonwinkler

Reply:
Yes, I would agree with that.

Reply:
I guess the supp delivered heating should also be 0.0 in calculating the cooling load introduced in the other equipment actuator? So it's going to introduce cooling loads and being resolved by ductless heat pump(or along with assumed integrated backup)?